core/src/main/java/org/apache/brooklyn/core/objs/proxy/InternalEntityFactory.java

```diff
@@ -369,6 +385,12 @@ public void run() {
                 }
                 ((AbstractEntity)entity).addLocations(spec.getLocations());
 
+                List<EntityInitializer> initializers = Stream.concat(spec.getInitializers().stream(), getDefaultInitializers().stream())
```

Comment:
this will run `spec.getInitializers()` _before_ default / global initializers, whereas i think default / globals should run first

also we have the lines below 394-396 which run `spec.getInitializers()` _again_ after

suggest this should just be `getDefaultInitializers().forEach(i -> i.apply(entity))` -- no need to concat streams

Reply:
Aaah I meant to remove line 394-396 and have only one loop to be DRY, hence the concatenation. I'll update @ahgittin

Reply:
no strong feelings about one loop or two, as long as globals done first, in case they want to fail fast to prevent even local initializers from running